Add citation section to GIS applications (#1721)
CSCfi/csc-user-guide: docs/apps/otb.md

!!! note
    It is also possible to run OTB in parallel in several computing nodes [using MPI](https://www.orfeo-toolbox.org/CookBook/CliInterface.html#parallel-execution-with-mpi) but it has not been tested yet on Puhti

## License 

> Orfeo ToolBox is licensed under the Apache License, Version 2.0. The full licence can be found [here](https://github.com/orfeotoolbox/OTB/tree/develop/Copyright)

Author: ```suggestion
Orfeo ToolBox is licensed under the Apache License, Version 2.0. The full licence can be found from the [Orfeo ToolBox github repository](https://github.com/orfeotoolbox/OTB/tree/develop/Copyright)
```

> Orfeo ToolBox is licensed under the Apache License, Version 2.0. The full licence can be found from the [Orfeo ToolBox github repository](https://github.com/orfeotoolbox/OTB/tree/develop/Copyright)

Reviewer: ```suggestion
Orfeo ToolBox is licensed under the Apache License, Version 2.0. [The full Orfeo ToolBox licence](https://github.com/orfeotoolbox/OTB/tree/develop/Copyright)
```

## Citation

> ```Grizonnet, M., Michel, J., Poughon, V. et al. Orfeo ToolBox: open source processing of remote sensing images. Open geospatial data, softw. stand. 2, 15 (2017). https://doi.org/10.1186/s40965-017-0031-6```

Reviewer: ```suggestion
`Grizonnet, M., Michel, J., Poughon, V. et al. Orfeo ToolBox: open source processing of remote sensing images. Open geospatial data, softw. stand. 2, 15 (2017). https://doi.org/10.1186/s40965-017-0031-6`
```
